### PR TITLE
Remove copypaste from StorageReplicatedMergeTree

### PR DESCRIPTION
**Pull request by tavplubix**
I hereby agree to the terms of the CLA available at: https://yandex.ru/legal/cla/?lang=en

Changelog category (leave one):
- Not for changelog (changelog entry is not required)

Detailed description / Documentation draft:
Was copy-pasted in #22971

### CODE REVIEW

File: src/Storages/MergeTree/IMergeTreeDataPart.h [hunk 1]
```diff
@@ -127,6 +127,7 @@ class IMergeTreeDataPart : public std::enable_shared_from_this<IMergeTreeDataPar
     void assertOnDisk() const;
 
     void remove(bool keep_s3 = false) const;
```
Comment (alexey-milovidov) [inline src/Storages/MergeTree/IMergeTreeDataPart.h:129]: There should not be any words about S3 in MergeTree.
We have similar logic for S3, HDFS and other decoupled storages.

Comment (tavplubix) [reply src/Storages/MergeTree/IMergeTreeDataPart.h:129]: I agree, but we already have S3-specific flags and methods in MergeTree and it's not obvious how to remove it.

Comment (tavplubix) [reply src/Storages/MergeTree/IMergeTreeDataPart.h:129]: We could rename it to more generic name like `keep_shared_data_in_decoupled_storage`, but "shared data" does not make sense for non-replicated MergeTree anyway.

File: src/Storages/MergeTree/IMergeTreeDataPart.h [hunk 2]
```diff
@@ -126,7 +126,7 @@ class IMergeTreeDataPart : public std::enable_shared_from_this<IMergeTreeDataPar
     /// Throws an exception if part is not stored in on-disk format.
     void assertOnDisk() const;
 
-    void remove(bool keep_s3 = false) const;
+    void remove() const;
 
     void projectionRemove(const String & parent_to, bool keep_s3 = false) const;
```
Comment (Avogar) [inline src/Storages/MergeTree/IMergeTreeDataPart.h:131]: Maybe rename keep_s3 here too for consistency?

**Review by Avogar — APPROVED** [review]

**Comment by tavplubix** [comment]
Yandex synchronization check - was broken in master
Performance - `The test 'group_by_sundy_li' is too slow to run as a whole.` - unrelated

Functional stateless tests (release, DatabaseReplicated) - 01154_move_partition_long:

clickhouse-server.log.gz
```
2021.06.11 18:51:56.836208 [ 97133 ] {b809ca1f-518a-4a38-b996-9093607134d3} <Debug> executeQuery: (from [::1]:55012, using production parser) (comment: '/usr/share/clickhouse-test/queries/0_stateless/01154_move_partition_long.sh') DROP TABLE IF EXISTS src_7
...
2021.06.11 18:52:28.126122 [ 97133 ] {b809ca1f-518a-4a38-b996-9093607134d3} <Error> executeQuery: Code: 159, e.displayText() = DB::Exception: Watching task /test/clickhouse/db/test_k6fzue/log/query-0000000097 is executing longer than distributed_ddl_task_timeout (=30) seconds. There are 1 unfinished hosts (0 of them are currently active), they are going to execute the query in background (version 21.7.1.7133) (from [::1]:55012) (comment: '/usr/share/clickhouse-test/queries/0_stateless/01154_move_partition_long.sh') (in query: DROP TABLE IF EXISTS src_7), Stack trace (when copying this message, always include the lines below):
```

clickhouse-server1.log.gz
```
2021.06.11 18:52:25.494472 [ 31100 ] {a68c341a-8617-450e-a542-8678dac68390} <Debug> executeQuery: (from 0.0.0.0:0, user: , using production parser) (comment: '/usr/share/clickhouse-test/queries/0_stateless/01154_move_partition_long.sh') /* ddl_entry=query-0000000097 */ DROP TABLE IF EXISTS test_k6fzue.src_7 NO DELAY
2021.06.11 18:52:25.494516 [ 31100 ] {a68c341a-8617-450e-a542-8678dac68390} <Trace> test_k6fzue.src_7 (ReplicatedMergeTreeRestartingThread): Restarting thread finished
2021.06.11 18:52:25.495500 [ 31100 ] {a68c341a-8617-450e-a542-8678dac68390} <Trace> test_k6fzue.src_7 (ReplicatedMergeTreeRestartingThread): Waiting for threads to finish
2021.06.11 18:52:25.495530 [ 31100 ] {a68c341a-8617-450e-a542-8678dac68390} <Information> test_k6fzue.src_7 (f6cbf141-a127-4979-b6cb-f141a127c979): Stopped being leader
2021.06.11 18:52:25.496339 [ 31100 ] {a68c341a-8617-450e-a542-8678dac68390} <Trace> test_k6fzue.src_7 (ReplicatedMergeTreeRestartingThread): Threads finished
2021.06.11 18:52:25.496354 [ 31100 ] {a68c341a-8617-450e-a542-8678dac68390} <Trace> test_k6fzue.src_7 (PartMovesBetweenShardsOrchestrator): PartMovesBetweenShardsOrchestrator thread finished
2021.06.11 18:52:25.496406 [ 31100 ] {a68c341a-8617-450e-a542-8678dac68390} <Trace> test_k6fzue.src_7 (f6cbf141-a127-4979-b6cb-f141a127c979): Found 13 old parts to remove.
2021.06.11 18:52:25.496417 [ 31100 ] {a68c341a-8617-450e-a542-8678dac68390} <Debug> test_k6fzue.src_7 (f6cbf141-a127-4979-b6cb-f141a127c979): Removing part from filesystem 0_0_14_2
2021.06.11 18:52:25.496711 [ 31100 ] {a68c341a-8617-450e-a542-8678dac68390} <Debug> test_k6fzue.src_7 (f6cbf141-a127-4979-b6cb-f141a127c979): Removing part from filesystem 0_0_60_11
2021.06.11 18:52:25.496971 [ 31100 ] {a68c341a-8617-450e-a542-8678dac68390} <Debug> test_k6fzue.src_7 (f6cbf141-a127-4979-b6cb-f141a127c979): Removing part from filesystem 0_0_103_19
2021.06.11 18:52:30.449152 [ 31100 ] {a68c341a-8617-450e-a542-8678dac68390} <Debug> test_k6fzue.src_7 (f6cbf141-a127-4979-b6cb-f141a127c979): Removing part from filesystem 1_0_12_2
2021.06.11 18:52:30.449463 [ 31100 ] {a68c341a-8617-450e-a542-8678dac68390} <Debug> test_k6fzue.src_7 (f6cbf141-a127-4979-b6cb-f141a127c979): Removing part from filesystem 1_0_59_10
2021.06.11 18:52:30.449742 [ 31100 ] {a68c341a-8617-450e-a542-8678dac68390} <Debug> test_k6fzue.src_7 (f6cbf141-a127-4979-b6cb-f141a127c979): Removing part from filesystem 3_35_71_7
2021.06.11 18:52:30.450008 [ 31100 ] {a68c341a-8617-450e-a542-8678dac68390} <Debug> test_k6fzue.src_7 (f6cbf141-a127-4979-b6cb-f141a127c979): Removing part from filesystem 5_2_20_3
2021.06.11 18:52:30.450273 [ 31100 ] {a68c341a-8617-450e-a542-8678dac68390} <Debug> test_k6fzue.src_7 (f6cbf141-a127-4979-b6cb-f141a127c979): Removing part from filesystem 5_2_62_11
2021.06.11 18:52:30.450548 [ 31100 ] {a68c341a-8617-450e-a542-8678dac68390} <Debug> test_k6fzue.src_7 (f6cbf141-a127-4979-b6cb-f141a127c979): Removing part from filesystem 7_0_12_2
2021.06.11 18:52:30.450801 [ 31100 ] {a68c341a-8617-450e-a542-8678dac68390} <Debug> test_k6fzue.src_7 (f6cbf141-a127-4979-b6cb-f141a127c979): Removing part from filesystem 7_0_64_11
2021.06.11 18:52:30.451092 [ 31100 ] {a68c341a-8617-450e-a542-8678dac68390} <Debug> test_k6fzue.src_7 (f6cbf141-a127-4979-b6cb-f141a127c979): Removing part from filesystem 8_7_54_9
2021.06.11 18:52:30.451360 [ 31100 ] {a68c341a-8617-450e-a542-8678dac68390} <Debug> test_k6fzue.src_7 (f6cbf141-a127-4979-b6cb-f141a127c979): Removing part from filesystem 9_0_13_2
2021.06.11 18:52:30.451619 [ 31100 ] {a68c341a-8617-450e-a542-8678dac68390} <Debug> test_k6fzue.src_7 (f6cbf141-a127-4979-b6cb-f141a127c979): Removing part from filesystem 9_0_80_14
2021.06.11 18:52:30.453251 [ 31100 ] {a68c341a-8617-450e-a542-8678dac68390} <Debug> DatabaseCatalog: Waiting for table f6cbf141-a127-4979-b6cb-f141a127c979 to be finally dropped
2021.06.11 18:52:30.522060 [ 31100 ] {a68c341a-8617-450e-a542-8678dac68390} <Debug> DDLWorker(test_k6fzue): Executed query: DROP TABLE IF EXISTS test_k6fzue.src_7 NO DELAY
2021.06.11 18:52:30.522084 [ 31100 ] {a68c341a-8617-450e-a542-8678dac68390} <Debug> MemoryTracker: Peak memory usage (for query): 0.00 B.
```

It took 5 seconds to remove part `0_0_103_19`... Need to adjust timeouts

**Pull request merged**